src/v1.js

```diff
@@ -215,6 +215,12 @@ api.declare({
   var hookId    = req.params.hookId;
   var hookDef   = req.body;
 
+  if (req.body.hookGroupId !== null && req.body.hookId !== null) {
```

Comment:
If not given, these will not be `null`, but undefined.  You can check for that with `if (req.body.hookGroupId)`.  You'll need to check the id's separately, too -- as written, if I pass just `hookGroupId`, this conditional will be false and the code to check the id's match will never execute!